storage/imxrt-flash/flashdrv.c

```diff
@@ -26,15 +27,109 @@
 #define QSPI_FREQ_133MHZ 0xc0000008
 
 
+#define QUAD_FAST_READ_SEQ_ID      0
+#define READ_STATUS_REG_SEQ_ID     1
+#define WRITE_ENABLE_SEQ_ID        3
+#define SECTOR_ERASE_SEQ_ID        5
+#define BLOCK_ERASE_SEQ_ID         8
+#define PAGE_PROGRAM_SEQ_ID        9
+#define CHIP_ERASE_SEQ_ID          11
+#define READ_JEDEC_ID_SEQ_ID       12
+
+
+static int flash_setWEL(flash_context_t *context, uint32_t dstAddr)
+{
+	int err = EOK;
+	flexspi_xfer_t xfer;
+
+	xfer.baseAddress = dstAddr;
+	xfer.operation = kFlexSpiOperation_Command;
+	xfer.seqId = WRITE_ENABLE_SEQ_ID;
+	xfer.seqNum = 1;
+	xfer.isParallelModeEnable = 0;
+
+	if ((err = flexspi_norFlashExecuteSeq(context->instance, &xfer)) != 0)
+		return -err;
+
+	return err;
+}
+
+
+static int flash_writeBytes(flash_context_t *context, uint32_t dstAddr, uint32_t *src, uint32_t size)
+{
+	int err = EOK;
+	flexspi_xfer_t xfer;
+
+	xfer.txSize = size;
+	xfer.txBuffer = src;
+	xfer.baseAddress = dstAddr;
+	xfer.operation = kFlexSpiOperation_Write;
+	xfer.seqId = PAGE_PROGRAM_SEQ_ID;
+	xfer.seqNum = 1;
+
+	if ((err = flexspi_norFlashExecuteSeq(context->instance, &xfer)) != 0)
+		return -err;
+
+	return err;
```

Comment:
return flexspi_norFlashExecuteSeq(context->instance, &xfer); would be simpler. Of course flexspi_norFlashExecuteSeq should return negative number in this case, or you can return -flexspi_norFlashExecuteSeq(context->instance, &xfer);

Applies to all other similar cases.